File: srcs/subscriber.py
import zmq

# Define the IP address and port to connect to
address = "tcp://193.206.157.36:9000"

# Create a ZeroMQ context
context = zmq.Context()

# Create a subscriber socket
subscriber = context.socket(zmq.SUB)

# Connect to the publisher
subscriber.connect(address)

# Subscribe to all messages (use "" for all, or use a specific topic)
subscriber.setsockopt_string(zmq.SUBSCRIBE, "")

print("Subscriber connected to {}. Waiting for messages...".format(address))

# Receive messages
while True:
    message = subscriber.recv_string()
    print("Received message:", message)

File: srcs/subscriber.cpp
#include <iostream>
#include <zmq.hpp>

int main() {
    // Create a context for the subscriber
    zmq::context_t context(1);

    // Create a subscriber socket
    zmq::socket_t subscriber(context, zmq::socket_type::sub);
    subscriber.connect("tcp://193.206.157.36:9000");

    // Subscribe to all messages
    subscriber.set(zmq::sockopt::subscribe, "");

    std::cout << "Subscriber connected to tcp://193.206.157.36:9000" << std::endl;

    while (true) {
        zmq::message_t message;
        subscriber.recv(message, zmq::recv_flags::none);
        std::cout << "Received message: " << message.to_string() << std::endl;
    }

    return 0;
}
